specs/ContextMenuRequested.md

```diff
@@ -0,0 +1,711 @@
+                }
+                wil::com_ptr<ICoreWebView2Deferral> deferral;
+                CHECK_FAILURE(args->GetDeferral(&deferral));
+                m_sampleWindow->RunAsync([deferral, showMenu]() {
+                    showMenu();
+                    CHECK_FAILURE(deferral->Complete());
+                });
+                return S_OK;
+            }).Get(),
+        &m_contextMenuRequestedToken);
+```
+## .Net/ WinRT Add or Remove Entries From Browser Menu 
+
+ ```c#
+    webView.CoreWebView2.ContextMenuRequested += delegate (object sender, CoreWebView2ContextMenuRequestedEventArgs args)
+    {
+        IList<CoreWebView2ContextMenuItem> menuList = args.MenuItems;
+        CoreWebView2ContextType context = args.ContextMenuParams.context;
+        args.Handled = false;
+        if (context == CoreWebView2ContextType.Image)
+        {
+            /// removes the last item in the collection
+            menuList.RemoveAt(menuList.Count - 1);
+        }
+        else if (context == CoreWebView2ContextType.Link)
+        {
+            /// add new item to end of colelction
+            CoreWebView2ContextMenuItem newItem = webView.CoreWebView2.Environment.CreateContextMenuItem(
+                1, "Display Link", "Shorcut", null, CoreWebView2ContextMenuItemKind.Normal, 0);
+            menuList.Insert(menuList.Count, newItem);
+        }
+    };
+    // CustomItemSelected event notified when a dev-inserted context menu item is selected
+    webView.CoreWebView2.CustomItemSelected += delegate (object sender, CoreWebView2CustomItemSelectedEventArgs args)
+    {
+        UInt32 customId = args.SelectedId;
+        CoreWebView2ContextMenuParams parameters = args.ContextMenuParams;
+        switch (customId)
+        {
+            case 1:
+                string linkUrl = parameters.LinkUrl;
+                MessageBox.Show(linkUrl, "Display Link", MessageBoxButton.YesNo);
+            break;
+        };
+    };
+``` 
+
+## .Net/ WinRT Use Data to Display Custom Context Menu 
+
+ ```c#
+    webView.CoreWebView2.ContextMenuRequested += delegate (object sender, CoreWebView2ContextMenuRequestedEventArgs args)
+    {
+        CoreWebView2Deferral deferral = args.GetDeferral();
+        System.Threading.SynchronizationContext.Current.Post((_) =>
+        {
+            using (deferral)
+            {
+                IList<CoreWebView2ContextMenuItem> menuList = args.MenuItems;
+                CoreWebView2ContextType context = args.ContextMenuParams.context;
+                args.Handled = true;
+                ContextMenu cm = this.FindResource("ContextMenu") as ContextMenu;
+                cm.Items.Clear();
+                for(int i = 0; i < menuList.Count; i ++){
+                    CoreWebView2ContextMenuItem current = menuList[i];
+                    MenuItem newItem = new MenuItem();
+                    newItem.Header = current.Name;
+                    newItem.Click += (s, ex) => args.SelectedItem = current;
```

Comment:
Do C# lambdas capture the value of current at the time the lambda is created or the variable current so that all the lambdas end up with current as its final value?

Reply:
I played with this using a message box and it seems like all the 'current' values used were at the time of lambda creation. I am not sure if this test directly translates to how it will act in the above scenario but it seemed like it captured them so they didn't end up with the same value.
What I did was: 
`                    for(int i = 0; i < arr.Length; i++)
                    {
                        string current = arr[i];
                        MenuItem newOne = new MenuItem();
                        newOne.Header = current;
                        newOne.Click += (s, ex) => MessageBox.Show(current, current, MessageBoxButton.YesNo);
                        cm.Items.Add(newOne);
                    }`